### PR TITLE
chore: Update release notes for [ TECH-1542 ] [TECH-1593 ] [ TECH-1594 ]

### PR DESCRIPTION
**Pull request by lucaCambi77**
Create a section for Database breaking changes for :

- https://dhis2.atlassian.net/browse/TECH-1542
- https://dhis2.atlassian.net/browse/TECH-1593
- https://dhis2.atlassian.net/browse/TECH-1594

### CODE REVIEW

File: releases/2.41/README.md [hunk 1]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+| programstageinstance        | event                          |
+| programstageinstancefilter  | eventfilter                    |
+| programstageinstancecomments| eventcomments                  |
+| programinstancecomments     | enrollmentcomments             |
+| trackedentityinstanceaudit  | trackedentityaudit             |
+| trackedentityinstancefilter | trackedentityfilter            |
+
+#### Columns Rename
+
+Likewise, some columns adapts to the new naming.
```
Comment (jbee) [inline releases/2.41/README.md:131]:
```suggestion
Likewise, some columns are adapted to the new naming. 
```

File: releases/2.41/README.md [hunk 2]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+| trackedentityinstanceaudit  | trackedentityaudit             |
+| trackedentityinstancefilter | trackedentityfilter            |
+
+#### Columns Rename
+
+Likewise, some columns adapts to the new naming. 
+
+Columns rename for deprecated `programstageinstance`
+
+| Table (existing or renamed) | Column Old Name                | Column New Name        |
```
Comment (jbee) [inline releases/2.41/README.md:135]:
```suggestion
| Table (new names) | Column Old Name                | Column New Name        |
```

**Review by jbee — APPROVED** [review]

File: releases/2.41/README.md [hunk 3]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
 
 * `/api/dataSummary` in the `objectCounts` object
 * `/api/system/objectCounts`
+
+## Database
+
+### Breaking Changes: tables and columns rename 
+
+Following Tracker's new [naming](#naming) in the API, we have some tables and columns renamed following the new convention.
```
Comment (teleivo) [inline releases/2.41/README.md:112]:
```suggestion
We have renamed some tables and columns following tracker's new [naming](#naming) in the API.
```

File: releases/2.41/README.md [hunk 4]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
 
 * `/api/dataSummary` in the `objectCounts` object
 * `/api/system/objectCounts`
+
+## Database
+
+### Breaking Changes: tables and columns rename
```
Comment (teleivo) [inline releases/2.41/README.md:110]:
```suggestion
### Breaking Changes: renamed tables and columns 
```

Comment (teleivo) [reply releases/2.41/README.md:110]: I would put this header under `### Tracker` like we did in `## API` 😄

File: releases/2.41/README.md [hunk 5]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+
+## Database
+
+### Breaking Changes: tables and columns rename 
+
+Following Tracker's new [naming](#naming) in the API, we have some tables and columns renamed following the new convention.
+
+Therefore, we align the database's names with the changes applied to `trackedEntityInstance`, `programInstance`, and `programStageInstance`.
+
+#### Tables Rename
```
Comment (teleivo) [inline releases/2.41/README.md:116]:
```suggestion
#### Renamed Tables
```

File: releases/2.41/README.md [hunk 6]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+| trackedentityinstance       | trackedentity                  |
+| programinstance             | enrollment                     |
+| programstageinstance        | event                          |
+| programstageinstancefilter  | eventfilter                    |
+| programstageinstancecomments| eventcomments                  |
+| programinstancecomments     | enrollmentcomments             |
+| trackedentityinstanceaudit  | trackedentityaudit             |
+| trackedentityinstancefilter | trackedentityfilter            |
+
+#### Columns Rename
```
Comment (teleivo) [inline releases/2.41/README.md:129]:
```suggestion
#### Renamed Columns
```

File: releases/2.41/README.md (hunk 1 above)
Comment (teleivo) [reply releases/2.41/README.md:131]: I think we can even remove any sentences in here. The why is already explained in `### Breaking Changes: tables and columns rename`. I don't think the table needs more explanation 😄

File: releases/2.41/README.md [hunk 7]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+
+Likewise, some columns adapts to the new naming. 
+
+Columns rename for deprecated `programstageinstance`
```
Comment (teleivo) [inline releases/2.41/README.md:133]:
```suggestion
The following `programstageinstance` related columns have been renamed
```

(same for the sentences below)

File: releases/2.41/README.md [hunk 8]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+
+#### Postgres Reference
+
+From Postgres docs for [alter table](https://www.postgresql.org/docs/current/sql-altertable.html)
+
+> The RENAME forms change the name of a table (or an index, sequence, view, materialized view, or foreign table), 
+> the name of an individual column in a table, or the name of a constraint of the table. 
+> When renaming a constraint that has an underlying index, the index is renamed as well. **There is no effect on the stored data.**
+
+Renaming a table or a table's column do not affect the data. For example, re-building a primary key index, which can be expensive for large tables, should not happen.
```
Comment (teleivo) [inline releases/2.41/README.md:183]:
```suggestion
Renaming a table or a table's column does not affect the data. For example, re-building a primary key index, which can be expensive for large tables, should not happen.
```

File: releases/2.41/README.md [hunk 9]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+Therefore, no downtime is expected following the migrations.
+
+We can check the transaction commit for the index creation hasn't changed after the renaming
```
Comment (teleivo) [inline releases/2.41/README.md:186]:
```suggestion
You can check that the index creation hasn't changed after the migration via the transaction commit.
```

File: releases/2.41/README.md [hunk 10]
```diff
@@ -104,3 +104,90 @@ in the following API response bodies
...
+
+```sql
+SELECT pg_xact_commit_timestamp(xmin)
+FROM pg_class
+WHERE relname = 'programstageinstance_pkey';
+```
+Notice if we want to run the query, Postgres needs to start with `-c track_commit_timestamp=on`
```
Comment (teleivo) [inline releases/2.41/README.md:193]:
```suggestion
Notice if you want to run the query, Postgres needs to start with `-c track_commit_timestamp=on`
```

**Review by teleivo — COMMENTED** [review]
Thanks a lot for documenting this @lucaCambi77 🥳

**Pull request merged**